Web3Swift/Transactions/EthTransactions.swift

```diff
@@ -7,13 +7,17 @@ import SwiftyJSON
 
 public final class EthTransactions: Transactions {
 
-    private let transactionsCountProcedure: RemoteProcedure
-    init(transactionsCountProcedure: RemoteProcedure) {
-        self.transactionsCountProcedure = transactionsCountProcedure
+    private let procedure: RemoteProcedure
+    init(network: Network, address: BytesScalar, blockChainState: BlockChainState) {
+        self.procedure = GetTransactionsCountProcedure(
+            network: network,
+            address: address,
+            blockChainState: blockChainState
+        )
     }
 
     public func count() throws -> NumberScalar {
-        let transactionsCountProcedure = self.transactionsCountProcedure
+        let transactionsCountProcedure = self.procedure
         return BigEndianCompactNumber(
             hex: SimpleString{
                 try transactionsCountProcedure.call()["result"].string()
```

Comment:
@biboran how can we remove `["result"]` from the code? Could we create something like `.value()` as a decorator for `JSON`?

Reply:
@rockfridrich we can get rid of `["result"]` duplication; however, we cannot make it just `.value()`. There are different types that can be represented by `JSON` (string, int etc) and sometimes there is a need to traverse `JSON` as a dictionary (`eth_getTransactionReceipt` call).

Reply:
@rockfridrich if you believe there is an issue with `["result"]` duplication make it a separate issue and we will discuss it there